cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1223,35 +1223,26 @@ private KafkaVersionChange getKafkaVersionChange(StatefulSet kafkaSts) {
         Future<ReconciliationState> getZookeeperDescription() {
             Promise<ReconciliationState> promise = Promise.promise();
 
-            vertx.createSharedWorkerExecutor("kubernetes-ops-pool").executeBlocking(
-                future -> {
-                    try {
-                        StatefulSet sts = zkSetOperations.get(namespace, ZookeeperCluster.zookeeperClusterName(name));
-                        Storage oldStorage = getOldStorage(sts);
+            zkSetOperations.getAsync(namespace, ZookeeperCluster.zookeeperClusterName(name)).setHandler(res -> {
+                if (res.succeeded()) {
+                    StatefulSet sts = res.result();
+                    Storage oldStorage = getOldStorage(sts);
 
-                        this.zkCluster = ZookeeperCluster.fromCrd(kafkaAssembly, versions, oldStorage);
+                    this.zkCluster = ZookeeperCluster.fromCrd(kafkaAssembly, versions, oldStorage);
 
-                        ConfigMap logAndMetricsConfigMap = zkCluster.generateMetricsAndLogConfigMap(zkCluster.getLogging() instanceof ExternalLogging ?
-                                configMapOperations.get(kafkaAssembly.getMetadata().getNamespace(), ((ExternalLogging) zkCluster.getLogging()).getName()) :
-                                null);
+                    ConfigMap logAndMetricsConfigMap = zkCluster.generateMetricsAndLogConfigMap(zkCluster.getLogging() instanceof ExternalLogging ?
+                            configMapOperations.getAsync(kafkaAssembly.getMetadata().getNamespace(), ((ExternalLogging) zkCluster.getLogging()).getName()).result() :
```

Comment:
You cannot call `result()` here, because that would need to wait after the future completes. The suggestion above should be able to deal with it.